lib_bgp_data/roas_parser/roas_automator.py

```diff
@@ -0,0 +1,191 @@
+            try:
+                count = table.get_count()
+                if count == 0:
+                    raise TableIsEmpty()
+            except (psycopg2.errors.UndefinedTable, TableIsEmpty):
+                if os.path.exists(self.prev_backup):
+                    table.clear_table()
+                    self._restore_table()
+                    table.create_index()
+            finally:
+                # Get the roas that need to be inserted into the table
+                roas = self._format_roas(self._get_json_roas())
+                _csv_dir = f"{self.csv_dir}/roas.csv"
+
+                # Try backing up the table at most three times
+                try:
+                    for _ in range(3):
+                        # Insert the rows into the table
+                        utils.clean_paths([self.path, self.csv_dir])
+                        utils.rows_to_db(roas, _csv_dir,
+                                         ROAs_Table, clear_table=False)
+                        table.create_index()
+
+                        # Create a backup, if it is successful, overwrite the
+                        # existing backup
+                        if self._backup_table():
+                            self._overwrite()
+                            break
+
+                        # If the backup is unsuccessful, restore the table
+                        # to the last backup
+                        if os.path.exists(self.prev_backup):
+                            table.clear_table()
+                            self._restore_table()
+                            table.create_index()
+
+                        # Enters this block if there is no previous backup,
+                        # need to make the table empty so that data can tried
+                        # to be inserted again
+                        else:
+                            table.execute("DELETE FROM roas")
+                            
+                    # If after three attempts the backups are unsuccessful,
+                    # then just insert the new roas and don't backup.
+                    else:
+                        utils.clean_paths([self.path, self.csv_dir])
+                        utils.rows_to_db(roas, _csv_dir,
+                                         ROAs_Table, clear_table=False)
+                        table.create_index()
+
+                        # Raise a backup error
+                        raise BackupError()
+
+                except subprocess.CalledProcessError as e:
+                    time = utils.now()
+                    subject = f"Roas Backup Error at {time}"
+                    body = ("There was an error backing up the roas table. "
+                            "Below are the outputs\n"
+                            f"STD_OUT Message: {e.stdout}\n"
+                            f"STD_ERR Message: {e.stderr}")
+                    utils.send_email(subject, body)
+
+                except BackupError:
```

Comment:
this should be combined with the above statement, and just change the body based on the error message. You can do this with except type1error, type2error etc as e I believe